server/image_pull.go

```diff
@@ -1,82 +1,38 @@
 package server
 
 import (
-	"errors"
-	"io"
-	"os"
-	"path/filepath"
-
 	"github.com/Sirupsen/logrus"
-	"github.com/containers/image/directory"
-	"github.com/containers/image/image"
-	"github.com/containers/image/transports"
+	"github.com/containers/image/copy"
+	"github.com/containers/image/signature"
+	"github.com/containers/image/types"
 	"golang.org/x/net/context"
 	pb "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 )
 
 // PullImage pulls a image with authentication config.
 func (s *Server) PullImage(ctx context.Context, req *pb.PullImageRequest) (*pb.PullImageResponse, error) {
 	logrus.Debugf("PullImage: %+v", req)
-	img := req.GetImage().GetImage()
-	if img == "" {
-		return nil, errors.New("got empty imagespec name")
-	}
-
-	// TODO(runcom): deal with AuthConfig in req.GetAuth()
-
-	// TODO(mrunalp,runcom): why do we need the SandboxConfig here?
-	// how do we pull in a specified sandbox?
-	tr, err := transports.ParseImageName(img)
-	if err != nil {
-		return nil, err
-	}
-	// TODO(runcom): figure out the ImageContext story in containers/image instead of passing ("", true)
-	src, err := tr.NewImageSource(nil, nil)
-	if err != nil {
-		return nil, err
-	}
-	i := image.FromSource(src)
-	blobs, err := i.BlobDigests()
-	if err != nil {
-		return nil, err
-	}
-
-	if err = os.Mkdir(filepath.Join(s.config.ImageDir, tr.StringWithinTransport()), 0755); err != nil {
-		return nil, err
+	// TODO(runcom?): deal with AuthConfig in req.GetAuth()
+	// TODO(somebody?): either rework PullImage to verify signatures, or do it by using PullImageUsingContexts, if that's enough
+	// TODO: what else do we need here? (Signatures when the story isn't just pulling from docker://)
+	image := ""
```

Comment:
should we check `image == nil` and error out if that's the case? instead of leaving to `RemoveImage`